src/utils/components/PendingChanges/utils/helpers.ts

```diff
@@ -43,18 +43,14 @@ export const checkCPUMemoryChanged = (
   if (isEmpty(vm) || isEmpty(vmi)) {
     return false;
   }
-  const vmRequests = vm?.spec?.template?.spec?.domain?.resources?.requests;
+  const vmMemory = vm?.spec?.template?.spec?.domain?.memory?.guest;
```

Comment:
I think we should add a selector for VMs and VMIs in `src/utils/resources/vm/utils/selectors.ts` and `src/utils/resources/vmi/utils/selectors.ts` respectively instead of accessing it this way throughout the code. It won't work everywhere, but it should cover most use cases and would make any future changes easier.

I'm sure there are other attributes that we're accessing this way as well, such as CPU, that would benefit from the same treatment. We can address them at appropriate points in the future.

Reply:
Yes and it is not time for cool refactorings, there is much more we could do here. But this is urgent bug we need to fix asap. Cool improvements later.

Reply:
Following DRY and other best practices aren't "cool refactorings". They are basic expectations for professional software developers and expected by most members of this team. We already have a pattern of using selectors for properties throughout the code base.

We shouldn't have allowed this to happen in the first place, but it did and now we need to fix it when and where it's appropriate. This is a good opportunity to fix this particular issue.

The change shouldn't take more than 10-15 minutes to make and I'm around all day to approve it.

Reply:
This PR should have already been merged!! it is blocking lots of stuff! One should think about DRY way much earlier than now, when copy pasting that code in our codebase.

Reply:
The bug has been present for some time now. When fixing it it's worth fixing it correctly. I said that we should have done better in the past, but there's nothing we can do about that now. The fact that we didn't do it perfectly in the past is not an excuse to do it poorly now. The goal is incremental improvement. It's a quick change and there's no good reason not to do it.

Reply:
> The bug has been present for some time now..

Yes, hidden behind 2 more bugs and was not urgent as it is now...

>  It's a quick change and there's no good reason not to do it.

So if it is such a quick change, why you simply didn't approved this PR and created another one with improvements you can see in the code? Obviously you are much better dev who can do things much faster...

I personally wouldn't underestimate refactoring, it always produces unnecessary bugs and that's what we really don't need now. Also it is not good to mix bug fixes with refactoring. I have very bad experience doing this.

Reply:
> Maybe I already could have an open followup refactoring PR if this was already merged. Because we already could be happy and finally calm that the urgent bug is gone and so more space for improvements would be here...

Create a second commit making the change. It doesn't need to be a separate PR. An individual commit can be reverted if it causes problems.

We're the only ones here today so it makes more sense to knock this out in a single PR and be done with it.